app/src/main/java/co/nimblehq/compose/crypto/ui/navigation/AppNavigation.kt

```diff
@@ -1,16 +1,38 @@
 package co.nimblehq.compose.crypto.ui.navigation
 
 import androidx.compose.runtime.Composable
+import androidx.compose.ui.res.stringResource
+import androidx.hilt.navigation.compose.hiltViewModel
 import androidx.navigation.*
 import androidx.navigation.compose.*
+import co.nimblehq.compose.crypto.R
+import co.nimblehq.compose.crypto.extension.collectAsEffect
+import co.nimblehq.compose.crypto.ui.common.AppDialogPopUp
+import co.nimblehq.compose.crypto.ui.screens.MainViewModel
 import co.nimblehq.compose.crypto.ui.screens.detail.DetailScreen
 import co.nimblehq.compose.crypto.ui.screens.home.HomeScreen
 
 @Composable
 fun AppNavigation(
     navController: NavHostController = rememberNavController(),
+    mainViewModel: MainViewModel = hiltViewModel(),
```

Comment:
Following MVVM architecture. The Navigator belongs on the View layer and should be isolated from the ViewModel layer. I don't think adding the ViewModel as a dependency is a good approach.

Please recheck the implementation on compose-sample repo: https://github.com/android/compose-samples/blob/2332f474336d8bbaf225dc00c845dde7c7cec300/Jetcaster/app/src/main/java/com/example/jetcaster/ui/JetcasterAppState.kt#L54-L57

Reply:
@hoangnguyen92dn Updated in https://github.com/nimblehq/jetpack-compose-crypto/pull/105/commits/2ea17a02b521333c33a75e919863253431a7de32 🙏

Reply:
@kaungkhantsoe The `AppNavigation` should not depend on the AppState and let's keep it neat. The main purpose of the Navigator class is to manage routing to the destinations and that's it. Observing the network status inside the Navigator class and navigating itself doesn't make sense.

Reply:
@kaungkhantsoe The initial idea about `AppNavigation.kt` is supposed to manage all routing in the app. Can we have a new Composable named `ComposeCryptoApp` and the `appState` + `appNagivation` as dependencies in stead?

Reply:
@hoangnguyen92dn added in https://github.com/nimblehq/jetpack-compose-crypto/pull/105/commits/7d58e3a04c36fb7e33992f20e524d17f472e5814. Please help me check if it is what you want. 🙏